CMakeLists.txt

```diff
@@ -27,7 +27,15 @@ endfunction()
 # DEPRECATED! Should be merged into add_module_library.
 function(enable_module target)
   if (MSVC)
-    set(BMI ${CMAKE_CURRENT_BINARY_DIR}/${target}.ifc)
+    set(BMI_DIR "${CMAKE_CURRENT_BINARY_DIR}/")
+    if (CMAKE_GENERATOR STREQUAL "Ninja")
+      file(RELATIVE_PATH BMI_DIR "${CMAKE_BINARY_DIR}" "${CMAKE_CURRENT_BINARY_DIR}")
+      if (BMI_DIR)
+        set(BMI_DIR "${BMI_DIR}/")
+      endif()
+      set(BMI_DIR "${BMI_DIR}CMakeFiles/${target}.dir/")
+    endif()
+    file(TO_NATIVE_PATH "${BMI_DIR}${target}.ifc" BMI)
```

Comment:
```suggestion
    if(CMAKE_GENERATOR STREQUAL "Ninja")
      # Ninja dyndep expects the .ifc output to be located in a specific relative path
      file(RELATIVE_PATH BMI_DIR "${CMAKE_BINARY_DIR}" "${CMAKE_CURRENT_BINARY_DIR}/CMakeFiles/${target}.dir")
    else()
      set(BMI_DIR "${CMAKE_CURRENT_BINARY_DIR}")
    endif()
    file(TO_NATIVE_PATH "${BMI_DIR}/${target}.ifc" BMI)
```

Reply:
Thanks, that's way cleaner! Tested and pushed.